.github/workflows/ci-chaincli.yml

```diff
@@ -0,0 +1,51 @@
+name: chaincli CI
+
+on:
+  push:
+    paths:
+      - "core/scripts/chaincli/**"
+  pull_request:
+    paths:
+      - "core/scripts/chaincli/**"
+
+jobs:
+  golangci:
+    name: chaincli-lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: "core/scripts/go.mod"
+          cache: true
+      - name: Touching core/web/assets/index.html
+        run: mkdir -p core/web/assets && touch core/web/assets/index.html
```

Comment:
Should we extract this into a composite action called "setup-x`? (Not sure what to name X, but its preparing for build). We use it in a bunch of places and it is kind of a magic string